Author: 
Description: Be more liberal in the types of DBM files we accept.
Index: apr-util-1.3.12/dbm/apr_dbm_berkeleydb.c
===================================================================
--- apr-util-1.3.12.orig/dbm/apr_dbm_berkeleydb.c	2010-04-26 21:43:11.000000000 +0400
+++ apr-util-1.3.12/dbm/apr_dbm_berkeleydb.c	2011-05-24 16:22:07.417491276 +0400
@@ -174,6 +174,21 @@
                                            apr_posix_perms2mode(perm))) != 0) {
                 /* close the DB handler */
                 (void) (*file.bdb->close)(file.bdb, 0);
+                if (dberr == EINVAL) {
+                   if ((dberr = db_create(&file.bdb, NULL, 0)) == 0) {
+                       if ((dberr = (*file.bdb->open)(file.bdb,
+#if DB_VER == 4
+                                               NULL,
+#endif
+                                               pathname, NULL,
+                                               DB_BTREE, dbmode,
+                                               apr_posix_perms2mode(perm))) != 0) {
+
+                            /* close the DB handler */
+                            (void) (*file.bdb->close)(file.bdb, 0);
+                       }
+		    }
+                }
             }
         }
         file.curs = NULL;
